indico/web/client/js/react/components/notes/NoteEditor.jsx

```diff
@@ -103,24 +108,50 @@ export function NoteEditor({apiURL, imageUploadURL, closeModal, getNoteURL}) {
       setCurrentInput('');
       getUserPreferences();
     }
-  }, [apiURL, getNoteURL]);
+  }, [getNoteURL, apiURL, setCurrentNoteRevision]);
 
-  const handleSubmit = async ({source}) => {
+  const handleSubmit = async ({source, forceRevision}) => {
     // getData is currently unused, keeping it around in case we need to enable lazyValue
     // for the TinyMCE editor widget in case larger notes cause performance issues that
     // can be resolved by only rendering them to an html string at submission time
     const currentValue = source.getData ? source.getData() : source;
+
     try {
       if (!currentValue) {
         await indicoAxios.delete(apiURL);
       } else {
-        await indicoAxios.post(apiURL, {
+        const resp = await indicoAxios.post(apiURL, {
           source: currentValue,
           render_mode: renderMode,
+          revision_id: forceRevision || currentNoteRevision.id,
         });
+        setCurrentNoteRevision(resp.data);
       }
     } catch (e) {
-      return handleSubmitError(e);
+      if (_.get(e, 'response.status') !== 418) {
+        return handleSubmitError(e);
+      }
+      // handle note conflict
+      await new Promise(resolve => {
+        injectModal(resolveModal => (
+          <ConflictModal
+            currentNote={currentValue}
+            externalNote={e.response.data.conflict}
+            onClose={() => {
+              resolve();
+              resolveModal();
+            }}
+            setCloseAndReload={setCloseAndReload}
+            overwriteChanges={async () => {
+              handleSubmit({
```

Comment:
You `await overwriteChanges` elsewhere, but here you don't await `handleSubmit`, so it doesn't actually wait for thr submit go finish.